File: app/src/main/java/com/ihadzhi/eatlimination/viewmodel/FoodsIncludedViewModel.java
package com.ihadzhi.eatlimination.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.ihadzhi.eatlimination.data.Diet;
import com.ihadzhi.eatlimination.data.EatliminationDatabase;
import com.ihadzhi.eatlimination.data.Food;

import java.util.List;

public class FoodsIncludedViewModel extends BaseViewModel {

    private LiveData<List<Food>> foods;
    private LiveData<Diet> activeDiet;

    public FoodsIncludedViewModel(@NonNull Application application) {
        super(application);
    }

    public LiveData<Diet> getActiveDiet() {
        if (activeDiet == null) {
            activeDiet = database.dietDao().fetchActiveDiet();
        }
        return activeDiet;
    }

    @Nullable
    public LiveData<List<Food>> getFoods(long dietId) {
        if (foods == null) {
            foods = new MutableLiveData<>();
            foods = database.foodDao().fetchByDietId(dietId);
        };
        return foods;
    }
}

File: app/src/main/java/com/ihadzhi/eatlimination/viewmodel/FoodsIncludedViewModel.kt
package com.ihadzhi.eatlimination.viewmodel

import android.app.Application
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.ihadzhi.eatlimination.data.Diet
import com.ihadzhi.eatlimination.data.Food
import java.util.concurrent.Executors

class FoodsIncludedViewModel(application: Application) : BaseViewModel(application) {

    private var foods: LiveData<List<Food>>? = null

    var activeDiet: LiveData<Diet>? = null
        get() {
            if (field == null) {
                field = database.dietDao().fetchActiveDiet()
            }
            return field
        }
        private set

    fun getFoods(dietId: Long): LiveData<List<Food>>? {
        if (foods == null) {
            foods = MutableLiveData()
            foods = database.foodDao().fetchByDietId(dietId)
        }
        return foods
    }

    fun removeFoodFromDiet(food: Food) {
        food.dietId = -1
        Executors.newSingleThreadScheduledExecutor().execute { database.foodDao().updateFood(food) }
    }
}